eis_toolkit/prediction/cnn_classification_and_probability.py

```diff
@@ -0,0 +1,417 @@
+    return normalized_data
+
+
+@beartype
+def make_one_hot_encoding(labels: np.ndarray):
+     """
+    Convert categorical labels to a one-hot encoded format.
+
+    This function transforms a list of numerical or categorical labels into a one-hot encoded matrix, which is a binary matrix representing the categorization of each label. One-hot encoding is commonly used in machine learning to handle categorical features.
+
+    Parameters:
+        - labels: A NumPy ndarray containing the labels to be encoded. Each label corresponds to a category or class.
+
+    Returns:
+        - label_encoded: A matrix where each row corresponds to a one-hot encoded representation of the input labels.
+
+    Raises:
+        - InvalidArgumentTypeException: Raised if the input 'labels' is None, indicating that no labels were provided for encoding.
+
+    """
+    if labels is None:
+        raise InvalidArgumentTypeException
+
+    # to categorical
+    enc = OneHotEncoder(handle_unknown="ignore")
+    # train and valid set
+    temp = np.reshape(labels, (-1, 1))
+    label_encoded = enc.fit_transform(temp).toarray()
+    return label_encoded
+
+
+@beartype
+def do_the_cnn(
+    input_shape_for_cnn: Union[tuple[int, int, int], tuple[int, int]],
+    convolution_kernel_size: tuple[int, int],
+    conv_list: list[int] = [8, 16],
+    pool_size: int = 2,
+    neuron_list: list[int] = [16],
+    dropout_rate: Union[None, float] = None,
+    last_activation: Literal["softmax", "sigmoid"] = "softmax",
+    regularization: Union[tf.keras.regularizers.L1, tf.keras.regularizers.L2, tf.keras.regularizers.L1L2, None] = None,
+    data_augmentation: bool = False,
+    optimizer: str = "Adam",
+    loss=Union[tf.keras.losses.BinaryCrossentropy, tf.keras.losses.CategoricalCrossentropy],
+    output_units=2,
+) -> tf.keras.Model:
+    """
+     Do an instance of CNN.
+
+    Parameters:
+       input_shape_for_cnn: shape of the input.
+       convolution_kernel_size: size of the kernel (usually is the last number of the input tuple)
+       pool_size: size of the pooling layer
+       conv_list: list of unit for the conv layers.
+       regularization: Type of regularization to insert into the CNN or MLP.
+       data_augmentation: if you want data augmentation or not (Random rotation is implemented).
+       optimizer: select one optimizer for the MLP.
+       loss: the loss function used to calculate accuracy.
+       neuron_list: List of unit or neuron used to build the network.
+       dropout_rate: if you want to use dropout add a number as floating point.
+       output_units: number of output classes.
+       last_activation: usually you should use softmax or sigmoid.
+
+    Return:
+        return the compiled model.
+
+    Raises:
+        InvalidArgumentTypeException: one of the arguments is invalid.
+        CNNException: raised when the input is not valid
+    """
+    # if regression and binary we can not uses more than 1
```

Comment:
No need for such a comment; rather, type out a message inside the exception. Look at other EIS Toolkit implementations for examples.